server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/step/PersistProjectLinksStep.java

```diff
@@ -89,17 +85,8 @@ private ProjectLinkVisitor(DbSession session) {
 
     @Override
     public void visitProject(Component project) {
```

Comment:
I may be wrong, but using the visitor pattern (DepthTraversalTypeAwareCrawler and TypeAwareVisitorAdapter) is overkill now that only project is relevant. I suggest to simply call use `treeRootHolder.getRoot()` to process links.